### ISSUE TITLE
[Question]: Suggestion on how to handle exception?

### ISSUE THREAD

**Issue by phil-lgr** [developit/workerize-loader#29]
With `async`/`await`, I tried to return a `new Error()` but discovered that Workers use the [Structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm).

Obviously returning a plain object (as below) works with some code to check if the `res` has a `.stack`.. 

```js
// worker.js
export async function expensive(time: number): Promise<number | Error> {
    try {
        const start = Date.now();
        let count: number = 0;
        while (Date.now() - start < time) {
            count++;
            if (count < 8561842) {
                throw new Error('OH NO');
            }
        }
        return count;
    } catch (error) {
        return {
            name: 'Exception in example.worker.ts',
            message: error.message,
            stack: error.stack
        };
    }
}

export default function exampleWorker() {
    return {
        expensive
    };
}
```

```js
// app.ts
import ExampleWorker from './workers/example.worker';

const instance = ExampleWorker();

instance.expensive(1000)
    .then(res => {
        if (res['stack']) {
            const error: Error = res as Error;
            throw {
                message: error.message,
                stack: error.stack
            };
        }
        return res;
    })
    .then(res => {
        console.log(`Ran ${res} loops`);
    })
    .catch(error => {
        console.log(error);
    });
```

I was wondering if someone had another idea to pass the exception 🤔

**Comment by phil-lgr** [comment]
Maybe the loader could serialize the exception and then re throw (aka call the error handler)?

**Comment by bboydflo** [comment]
@phil-lgr have you found a solution to this?

**Comment by developit** [comment]
You can throw a String, but I agree we should fix this internally by serializing and recreating the Error.

Here's what workerize (the non-webpack version) does:
https://github.com/developit/workerize/blob/master/src/index.js#L68-L75

**Comment by phil-lgr** [comment]
Ok cool I will have a look at this!

**Comment by phil-lgr** [comment]
PR #33

**Comment by developit** [comment]
Merged and released as 1.0.3!
